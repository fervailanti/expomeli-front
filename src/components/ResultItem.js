import React from 'react'

const ResultItem = ({ image, title, price, action, freeShipping, condition }) => (
  <div className="item">
    <img src={image} className="img" alt="product" onClick={action} />
    <div className="main">
      <p className="condition">{condition}</p>
      <p className="title" onClick={action}>
        {title}
      </p>
      <p className="price">{price}</p>
      {freeShipping && <p className="tag">Env√≠o gratis</p>}
    </div>
  </div>
)

export default ResultItem
