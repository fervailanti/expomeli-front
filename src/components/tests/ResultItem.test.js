import React from 'react'
import ResultItem from '../ResultItem'
import { render } from '@testing-library/react'

it('ResultItem component should render correctly', () => {
  const tree = render(
    <ResultItem
      image="https://picsum.photos/500"
      title="iPhone 11"
      price="$ 179.499,99"
      action={() => null}
      condition="Nuevo"
      freeShipping
    />
  )
  expect(tree.container).toMatchSnapshot()
})
