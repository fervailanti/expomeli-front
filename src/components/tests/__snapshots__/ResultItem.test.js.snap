// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem component should render correctly 1`] = `
<div>
  <div
    class="item"
  >
    <img
      alt="product"
      class="img"
      src="https://picsum.photos/500"
    />
    <div
      class="main"
    >
      <p
        class="condition"
      >
        Nuevo
      </p>
      <p
        class="title"
      >
        iPhone 11
      </p>
      <p
        class="price"
      >
        $ 179.499,99
      </p>
      <p
        class="tag"
      >
        Env√≠o gratis
      </p>
    </div>
  </div>
</div>
`;
